bentoml/cli/__init__.py

```diff
@@ -420,6 +421,44 @@ def install_completion(append, shell, path):
                 "install it with `pip install click_completion`"
             )
 
+    @bentoml_cli.command(
+        help='Retrieves BentoService artifacts into a target directory',
+        short_help="Retrieves BentoService artifacts into a target directory",
+    )
+    @conditional_argument(pip_installed_bundle_path is None, "bento", type=click.STRING)
+    @click.option(
+        '--target_dir',
+        help="Directory to put artifacts into. Defaults to pwd.",
+        default=os.getcwd(),
+    )
+    def retrieve(bento, target_dir):
+        if ':' not in bento:
+            _echo('BentoService {bento} invalid - specify name:version')
+            return
+        name, version = bento.split(':')
+
+        yatai_client = YataiClient()
+
+        track_cli('bento-retrieve')
+        get_bento_result = yatai_client.repository.get(name, version)
+        if get_bento_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                get_bento_result.status
+            )
+            raise BentoMLException(
+                f'BentoService {name}:{version} not found - '
+                f'{error_code}:{error_message}'
+            )
+            return
+
+        bento_service_bundle_path = resolve_bundle_path(
```

Comment:
After moving to `bento.py`, this can actually get rid of `resolve_bundle_path ` and use the `get_bento_result ` directly:

```
            if get_bento_result.bento.uri.s3_presigned_url:
                # Use s3 presigned URL for downloading the repository if it is presented
                return get_bento_result.bento.uri.s3_presigned_url
            else:
                return get_bento_result.bento.uri.uri
```

resolve_bundle_path also resolved when `bento` arg is a local path(e.g. `bentoml serve $DIR` vs `bentoml serve IrisClassifier:latest`), which is not applicable here